tests/drivers/i2c/i2c_target_api/testcase.yaml

```diff
@@ -32,3 +32,10 @@ tests:
       - nucleo_f091rc
     extra_configs:
       - CONFIG_APP_DUAL_ROLE_I2C=y
+  drivers.i2c.target_api.single_role:
+     platform_allow:
+       - frdm_mcxn947/mcxn947/cpu0
```

Comment:
for this board, you need also add overlay like what is existing in boards folder